app/views/layouts/_topnav.html.haml

```diff
@@ -6,36 +6,41 @@
   %div#topNavigationToggler.navbar-collapse.collapse.justify-content-between
     %ul.navbar-nav
       %li.nav-item
-        = link_to("Browse", ontologies_path(), class: "nav-link")
+        = link_to(t(:_browse), ontologies_path(), class: "nav-link")
       %li.nav-item
-        = link_to("Search", "/search", class: "nav-link")
+        = link_to(t(:_search), "/search", class: "nav-link")
       %li.nav-item
-        = link_to("Mappings", mappings_path(), class: "nav-link")
+        = link_to(t(:_mappings), mappings_path(), class: "nav-link")
       %li.nav-item
-        = link_to("Recommender", recommender_index_path, class: "nav-link")
+        = link_to(t(:_recommender), recommender_index_path, class: "nav-link")
       %li.nav-item
-        = link_to("Annotator", annotator_index_path(), class: "nav-link")
+        = link_to(t(:_annotator), annotator_index_path(), class: "nav-link")
       - if $NCBO_ANNOTATORPLUS_ENABLED == true
         %li.nav-item
-          = link_to("NCBO Annotator+", '/ncbo_annotatorplus', class: "nav-link")
+          = link_to(t(:_ncbo_annotator_plus), '/ncbo_annotatorplus', class: "nav-link")
       %li.nav-item
-        = link_to("Projects", projects_path(), class: "nav-link")
+        = link_to(t(:_projects), projects_path(), class: "nav-link")
       %li.nav-item
-        = link_to("Landscape", '/landscape', class: "nav-link")
+        = link_to(t(:_landscape), '/landscape', class: "nav-link")
       -if (!session[:user].nil? && session[:user].admin?)
         %li.nav-item
-          = link_to("Admin", admin_index_path, class: "nav-link")
+          = link_to(t(:_admin), admin_index_path, class: "nav-link")
     %ul.navbar-nav
+      %li.nav-item{ style: "margin-right: 10px;" }
+        %select#language-select.form-control{ data: { controller: "language-change", action: "change->language-change#setLocale"} }
```

Comment:
you should use here a variable from the controller, the map {en: 'english' ,... fr: '}

Reply:
more details please